### PR TITLE
lsp.py: Fix various problems with subdirectory edge cases

### PR DESCRIPTION
**Pull request by Marenz**
* fix properly path parsing with nested subdirectories
* fix regex to be able to handle `file://..`
* fix not checking all expected diagnostics
* Some extra comments and renames

### CODE REVIEW

File: test/lsp.py [hunk 1]
```diff
@@ -586,6 +591,12 @@ def test_diagnostics(self):
                         marker=markers[expected_diagnostic.marker]
                     )
 
+            if len(expected_diagnostics_per_file) > 0:
+                raise ExpectationFailed(
+                    f"Expected but received no diagnostics for {expected_diagnostics_per_file}",
```
Comment (christianparpart) [inline test/lsp.py]: "Expected but received no diagnostics" sounds rather strange to me, grammer-wise. Did you miss something in this sentence?

File: test/lsp.py [hunk 2]
```diff
@@ -889,22 +900,31 @@ def wait_for_diagnostics(self, solc: JsonRpcProcess) -> List[dict]:
 
         return sorted(reports, key=lambda x: x['uri'])
 
+    def normalizeUri(self, uri, no_suffix = True):
+        relUri = uri.replace(self.project_root_uri + "/", "")
```
Comment (christianparpart) [inline test/lsp.py]:
```suggestion
        relativeURI = uri.replace(self.project_root_uri + "/", "")
```

**Review by christianparpart — CHANGES_REQUESTED** [review]
thanks. also the naming changes come quite in handy. much more self-explanatory :)

File: test/lsp.py (hunk 1 above)
Comment (Marenz) [reply test/lsp.py]: "*grammar" your mean :P

 I don't see a problem with the sentence.. "[We] expected[, ]but received no diagnostics for .. "

File: test/lsp.py [hunk 3]
```diff
@@ -889,22 +900,31 @@ def wait_for_diagnostics(self, solc: JsonRpcProcess) -> List[dict]:
 
         return sorted(reports, key=lambda x: x['uri'])
 
+    def normalizeUri(self, uri, no_suffix = True):
```
Comment (christianparpart) [inline test/lsp.py]: Second parameter is **never** used so it seems? Did I overlook it?

Comment (Marenz) [reply test/lsp.py]: Ah, there was a function I wanted to use it but then decided otherwise

Comment (christianparpart) [reply test/lsp.py]: then probably clean this argument up and not have that if-statement below.

**Review by christianparpart — APPROVED** [review]

**Review by christianparpart — CHANGES_REQUESTED** [review]
`lsp.py: error: unrecognized arguments: --batch` needs a fix

File: test/lsp.py [hunk 4]
```diff
@@ -272,27 +273,34 @@ def create_cli_parser() -> argparse.ArgumentParser:
...
         dest="fail_fast",
         action="store_true",
         help="Terminates the running tests on first failure."
     )
+    parser.set_defaults(batch=False)
+    parser.add_argument(
+        "-b", "--batch",
+        dest="batch",
+        action="store_true",
+        help="Prevent interactive queries and just fail instead."
```
Comment (christianparpart) [inline test/lsp.py:286]: Honestly, is `--batch` then the right word to go with? Why not `--non-interactive` to avoid interactive use (such as in CI runs)?

**Review by christianparpart — APPROVED** [review]

**Pull request merged**